src/vscode-bicep/src/panes/deploy/app/components/sections/ParametersInputView.tsx

```diff
@@ -0,0 +1,48 @@
+import { VSCodeTextField, VSCodeCheckbox, VSCodeButton, VSCodeDivider } from "@vscode/webview-ui-toolkit/react";
+import { FC } from "react";
+import { ParamData, ParamDefinition, ParametersMetadata, TemplateMetadata } from "../models";
+import { ParamInputBox } from "../ParamInputBox";
+
+interface ParametersInputViewProps {
+  template?: TemplateMetadata;
+  parameters?: ParametersMetadata;
+  disabled: boolean;
+  onValueChange: (name: string, data: ParamData) => void;
+  onEnableEditing: () => void;
+  onPickParametersFile: () => void;
+}
+
+export const ParametersInputView: FC<ParametersInputViewProps> = ({ template, parameters, disabled, onValueChange, onEnableEditing, onPickParametersFile }) => {
+  if (!template || !parameters) {
+    return null;
+  }
+
+  const { parameterDefinitions } = template;
+  const { sourceFilePath } = parameters;
+
+  return (
+    <section className="form-section">
+      <VSCodeDivider />
+      <h2>Parameters</h2>
+      {sourceFilePath ? <VSCodeTextField value={sourceFilePath} disabled={true}>File Path</VSCodeTextField> : null}
```

Comment:
IIRC this should work:
```suggestion
      {sourceFilePath && <VSCodeTextField value={sourceFilePath} disabled={true}>File Path</VSCodeTextField>}
```
Same for the components below.

Reply:
We can replace the ternary operators in `ResultsView.tsx` as well.